rnaseq-cgl-pipeline/wrapper.py

```diff
@@ -1,102 +1,203 @@
+    # If no arguments provided, print full help menu
+    if len(sys.argv) == 1:
+        parser.print_help()
+        sys.exit(1)
+    # Get name of most recent running container. If socket is mounted, should be this one.
+    try:
+        name = subprocess.check_output(['docker', 'ps', '--format', '{{.Names}}']).split('\n')[0]
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError('No container detected, ensure Docker is being run with: '
+                           '"-v /var/run/docker.sock:/var/run/docker.sock" as an argument. \n\n{}'.format(e.message))
     # Get name of mounted volume
     blob = json.loads(subprocess.check_output(['docker', 'inspect', name]))
     mounts = blob[0]['Mounts']
     # Ensure docker.sock is mounted correctly
     sock_mount = [x['Source'] == x['Destination'] for x in mounts if 'docker.sock' in x['Source']]
-    if len(sock_mount) != 1:
-        raise IllegalArgumentException('Missing socket mount. Requires the following:'
-                                       'docker run -v /var/run/docker.sock:/var/run/docker.sock')
+    require(len(sock_mount) == 1, 'Missing socket mount. Requires the following: '
+                                  'docker run -v /var/run/docker.sock:/var/run/docker.sock')
     # Ensure formatting of command for 2 mount points
     if len(mounts) == 2:
-        if not all(x['Source'] == x['Destination'] for x in mounts):
-            raise IllegalArgumentException('Docker Src/Dst mount points, invoked with the -v argument,'
-                                           'must be the same if only using one mount point aside from the '
-                                           'docker socket.')
+        require(all(x['Source'] == x['Destination'] for x in mounts),
+                'Docker Src/Dst mount points, invoked with the -v argument, '
+                'must be the same if only using one mount point aside from the docker socket.')
         work_mount = [x['Source'] for x in mounts if 'docker.sock' not in x['Source']]
     else:
         # Ensure only one mirror mount exists aside from docker.sock
         mirror_mounts = [x['Source'] for x in mounts if x['Source'] == x['Destination']]
         work_mount = [x for x in mirror_mounts if 'docker.sock' not in x]
-        if len(work_mount) > 1:
-            raise IllegalArgumentException('Too many mirror mount points provided, see documentation.')
-        if len(work_mount) == 0:
-            raise IllegalArgumentException('No required mirror mount point provided, see documentation.')
-    # Enforce file input standards
+        require(len(work_mount) == 1, 'Wrong number of mirror mounts provided, see documentation.')
+    # Look for inputs in work directory
+    star = glob(os.path.join(work_mount[0], 'star*'))
+    rsem = glob(os.path.join(work_mount[0], 'rsem*'))
+    kallisto = glob(os.path.join(work_mount[0], 'kallisto*'))
+    if not args.star:
```

Comment:
Unless I'm misunderstanding @JakeNarkizian, please don't do that, since that would mean that you're hardcoding wrapper script paths. I think it's fine to give people recommended paths in documentation, but I dislike hardcoding things.

As for:

> I was told to simplify the number of necessary arguments needed to invoke the container, and that passing in paths to the inputs was overwhelming for some users.

I know this isn't your decision, @jvivian, but I think this suggestion is a mistake. If we don't expect our users to be sufficiently sophisticated to pass input paths, why do we expect them to be sufficiently sophisticated to properly set the Docker volume mounts so that CWD mounts properly and the files are automagically discovered? IMO, it is vastly preferable to have users pass input paths that we can easily validate and throw a sane error for, instead of doing something automagical. I see very little upside to this automagical approach, and lots of downside.